### ISSUE TITLE
Impossible to restore packages if a feed in nuget.config contains a parenthesis in its name

### ISSUE THREAD

**Issue by actionthomas** [microsoft/azure-pipelines-tasks#14936]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DotNetCoreCLI

## Environment
- Server - Azure Pipelines or TFS on-premises?
TFS On-Prem (Version Azure DevOps Server 2020 Update 1 RC2)

- Agent - Hosted or Private: 
 Private agent (2.186.1 on windows)

## Issue Description
When running dotnet restore with a feed whose name contains a non alphanumeric character in the nuget.config file, restore fails.

### Task logs
[task.log](https://github.com/microsoft/azure-pipelines-tasks/files/6609263/task.log)

### Error logs

` "E:\_w\2\407\s\Sources\Socle.Teleconsultation.ProvidersTests\Socle.Teleconsultation.ProvidersTests.csproj" (Restore cible) (1) ->
       "E:\_w\2\407\s\Sources\Socle.Teleconsultation.ProvidersTests\Socle.Teleconsultation.ProvidersTests.csproj" 
(_GenerateRestoreGraphProjectEntry cible) (1:5) ->
         C:\Program Files\dotnet\sdk\5.0.300\NuGet.targets(565,5): error : NuGet.Config n'est pas un fichier XML valide. Chemin : 'E:\_w\2\407\Nuget\tempNuGet_123306.config'. 
       C:\Program Files\dotnet\sdk\5.0.300\NuGet.targets(565,5): error :   The '(' character, hexadecimal value 0x28, cannot be included in a name. Line 24, position 46. 
`

**Comment by tejasd1990** [comment]
Hi @actionthomas ,
can you share a sample nuget.config file contanining the paranthesie?

**Comment by actionthomas** [comment]
Hi !
Here it is (I have to change some data because it's public in here)
[Nuget.Config](https://github.com/microsoft/azure-pipelines-tasks/files/6622282/Nuget.Config.txt)

If you need the original one I'll have to send it to you privately :)

**Comment by tejasd1990** [comment]
@actionthomas ,
no this is good enough. 
I was able to repro this issue, even with running cli commands myself from a shell, so this is not a task issue.
The error comes because xml tags cannt contain such special cahracters. Can you check [this](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials) example, and make changes to your nugetconfig file to unblock yourselves? In the doc example, they have substituted for space, you can do similarly for paramtheses.

**Comment by actionthomas** [comment]
Yes I removed the parenthesis yesterday to be able to build :)

**Comment by tejasd1990** [comment]
ok, glad to know that. Since this is not a task issue, closing, thanks.
